src/main/java/org/moltimate/moltimatebackend/service/AlignmentService.java

```diff
@@ -67,11 +68,10 @@ public ActiveSiteAlignmentResponse alignActiveSites(ActiveSiteAlignmentRequest a
                 results.get(structure.getPDBCode())
                         .addAll(motifs.stream()
                                 .parallel()
-                                .map(motif -> alignActiveSites(
-                                        structure,
-                                        motif,
-                                        precision
-                                ))
+                                .map(motif -> {
+                                    Structure motifStructure = ProteinUtils.queryPdb(motif.getPdbId());
+                                    return alignActiveSites(structure, motif, motifStructure, precision);
```

Comment:
Previously we were looking up the motif structure down at the RMSD level, this is not sufficient for the case of a custom motif with a custom structure. By looking it up/passing it in here we can handle custom structures as well (as seen in the below case)

Reply:
👍